### DISCUSSION TITLE
Is there an option for cancelling a query if it's already in flight?

### DISCUSSION THREAD

**Discussion by stevenschmatz** [TanStack/query#1299] [Q&A]
Basically what I'm looking for is a shorter version of this:

```tsx
function runPostRequest() {}

function Component() {
  const [makeRequest, { isLoading }] = useMutation(runPostRequest);

  const makeRequestOnce = () => {
    if (!isLoading) {
      makeRequest();
    }
  }
}
```

Is there any shorthand for this, so the `makeRequestOnce` function would not be necessary? Ideally something like:

```tsx
function runPostRequest() {}

function Component() {
  const [makeRequestOnce, { isLoading }] = useMutation(runPostRequest, { singleInstance: true });
}
```

Thanks in advance.

**Comment by TkDodo** [comment]
Hmm, I'm a but confused. are you talking about `queries` or `mutations` now? mutation are imperative and a side-effect, so you should probably not execute them in the render function like you are doing in the above code. its more like: you click a button, you call the `mutate` function. Otherwise, it will be executed every render, and that's not good.

So, I have to ask: what's your use-case?
